src/DynamicData.Tests/Cache/BindAndSortFixture.cs

```diff
@@ -0,0 +1,372 @@
+
+        firstItem.Should().Be(insert);
+    }
+
+    [Fact]
+    public void AppendAtEnd()
+    {
+        var people = _generator.Take(100).ToArray();
+        _source.AddOrUpdate(people);
+
+        var toInsert = new Person("zzzzz", 1000);
+
+        _source.AddOrUpdate(toInsert);
+
+        _boundList.Count.Should().Be(101);
+        
+        var last = _boundList[^1];
+        last.Should().Be(toInsert);
+
+    }
+
+    [Fact]
+    public void AppendInMiddle()
+    {
+        _source.AddOrUpdate(Enumerable.Range(0,100).Select(i=> new Person($"P{i}",i)));
+
+        //create age 0 to ensure it is inserted first
+        var insert = new Person("Marvin", 50);
+
+        _source.AddOrUpdate(insert);
+
+        _boundList.Count.Should().Be(101);
+
+        var index = _boundList.IndexOf(insert);
+
+        index.Should().Be(50);
+    }
+
+    [Fact]
+    public void BatchOfVariousChanges()
+    {
+        var people = Enumerable.Range(0, 100).Select(i => new Person($"P{i}", i)).ToArray();
+        _source.AddOrUpdate(people);
+
+        var toupdate = people[3];
+
+        // mixture of add, updates and removed
+        _source.Edit(innerCache =>
+        {
+            innerCache.Remove(people[0].Key);
+            innerCache.AddOrUpdate(new Person("Mr", "Z", 50, "M"));
+            innerCache.Remove(people[1].Key);
+            innerCache.AddOrUpdate(new Person(toupdate.Name, toupdate.Age - 24));
+            innerCache.Remove(people[7].Key);
+        });
+
+        // mirror actions to caches
+        var expected = Enumerable.Range(0, 100).Select(i => new Person($"P{i}", i)).ToList();
+        expected.RemoveAt(IndexFromKey(people[0].Key));
+        expected.RemoveAt(IndexFromKey(people[1].Key));
+        expected.Add(new Person("Mr", "Z", 50, "M"));
+        expected[IndexFromKey(toupdate.Key)] = new Person(toupdate.Name, toupdate.Age - 24);
+        expected.RemoveAt(IndexFromKey(people[7].Key));
+
+        int IndexFromKey(string key) => expected.FindIndex(p => p.Key == key);
+
+        var expectedInOrder = expected.OrderBy(p => p, _comparer).ToList();
+
+        expectedInOrder.SequenceEqual(_boundList).Should().BeTrue();
+    }
+
+
+    [Fact]
+    public void BatchOfVariousEndingInClear()
+    {
+        var people = _generator.Take(10).ToArray();
+        _source.AddOrUpdate(people);
+
+        _source.Edit(updater =>
+            {
+                updater.Clear();
+                updater.AddOrUpdate(_generator.Take(10).ToArray());
+                updater.Clear();
+            });
+
+        _boundList.Count.Should().Be(0);
+
+    }
+
+
+    [Fact]
+    public void LargeBatchChange()
+    {
+        // this should produce what are effectively 2 resets for the bound collection
+        _source.AddOrUpdate(Enumerable.Range(0, 100).Select(i => new Person($"P{i}", i)));
+        _source.AddOrUpdate(Enumerable.Range(100, 100).Select(i => new Person($"P{i}", i)));
+
+
+        _boundList.Count.Should().Be(200);
```

Comment:
In my mind, we should always be running the same assertions on the results, I.E. assert that the bound list matches the source items, when sorted. It's not like there's anything to be gained by doing just a count assertion here.